spring-kafka/src/main/java/org/springframework/kafka/listener/BatchConsumerAwareMessageListener.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.kafka.listener;
+
+import java.util.List;
+
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+
+import org.springframework.kafka.support.Acknowledgment;
+
+/**
+ * Listener for handling a batch of incoming Kafka messages; the list
+ * is created from the consumer records object returned by a poll.
+ * Access to the {@link Consumer} is provided.
+ *
+ * @param <K> the key type.
+ * @param <V> the value type.
+ *
+ * @author Gary Russell
+ *
+ * @since 2.0
+ */
+@FunctionalInterface
+public interface BatchConsumerAwareMessageListener<K, V> extends BatchAcknowledgingMessageListener<K, V> {
```

Comment:
Yes, I think I can do that with how things worked out in the end; this was an artifact of an earlier iteration.